src/Otp/GoogleAuthenticator.php

```diff
@@ -61,10 +66,10 @@ public static function getKeyUri($type, $label, $secret, $counter = null, $optio
         }
 
         // This is the base, these are at least required
-        $otpauth = 'otpauth://' . $type . '/' . $label . '?secret=' . $secret;
+        $otpauth = 'otpauth://' . $type . '/' . str_replace(array(':', ' '), array('%3A', '%20'), $label) . '?secret=' . rawurlencode($secret);
 
         if ($type == 'hotp' && !is_null($counter)) {
-            $otpauth .= '&counter=' . $counter;
+            $otpauth .= '&counter=' . rawurlencode($counter);
```

Comment:
Counter is an integer. A check for integer (or forced conversion) would be better then urlencode which does nothing to a number like that.